File: leetcode.com/python/71_Simplify_Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        if not path:
            return path

        stack = []
        for directory in path.split("/"):
            if directory == "..":
                if stack:
                    stack.pop()
            elif directory == "." or not directory:
                continue
            else:
                stack.append(directory)

        finalDirectoryPath = "/" + "/".join(stack)
        return finalDirectoryPath

File: leetcode.com/swift/DS_ALGO_PRAC.playground/Sources/swift/71_Simplify_Path.swift
import Foundation

class Solution {
    func simplifyPath(_ path: String) -> String {
        var stack = [String]()
        let pathComponent = path.split(separator: "/")
        for component in pathComponent {
            if component == "" || component == "." || (stack.isEmpty && component == "..") {
                continue
            }
            if component == ".." {
                stack.removeLast()
                continue
            }
            stack.append(String(component))
        }
        return stack.isEmpty ? "/" : stack.reduce("") { $0 + "/" + $1 }
    }
}
